File: spring-data-requery/src/main/java/org/springframework/data/requery/converters/Java8Times.java
package org.springframework.data.requery.converters;

import lombok.experimental.UtilityClass;
import org.jetbrains.annotations.NotNull;

import java.time.format.DateTimeFormatter;
import java.time.temporal.TemporalAccessor;

/**
 * org.springframework.data.requery.converters.Java8Times
 *
 * @author debop
 */
@UtilityClass
public class Java8Times {

    public static long NANO_IN_SECONDS = 1_000_000_000L;

    @NotNull
    public static String toIsoOffsetDateTimeString(@NotNull TemporalAccessor accessor) {
        return DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(accessor);
    }

    @NotNull
    public static String toIsoOffsetDateString(@NotNull TemporalAccessor accessor) {
        return DateTimeFormatter.ISO_OFFSET_DATE.format(accessor);
    }

    @NotNull
    public static String toIsoOffsetTimeString(@NotNull TemporalAccessor accessor) {
        return DateTimeFormatter.ISO_OFFSET_TIME.format(accessor);
    }

    @NotNull
    public static String toIsoZonedDateTimeString(@NotNull TemporalAccessor accessor) {
        return DateTimeFormatter.ISO_ZONED_DATE_TIME.format(accessor);
    }
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/converters/Java8Times.kt
package org.springframework.data.requery.converters

import java.time.format.DateTimeFormatter
import java.time.temporal.TemporalAccessor

/**
 * org.springframework.data.requery.converters.Java8Times
 * @author debop
 * @since 18. 7. 2
 */


fun TemporalAccessor.toIsoOffsetDateTimeString(): String = DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(this)

fun TemporalAccessor.toIsoOffsetDateString(): String = DateTimeFormatter.ISO_OFFSET_DATE.format(this)

fun TemporalAccessor.toIsoOffsetTimeString(): String = DateTimeFormatter.ISO_OFFSET_TIME.format(this)

fun TemporalAccessor.toIsoZonedDateTimeString(): String = DateTimeFormatter.ISO_ZONED_DATE_TIME.format(this)